tests/naive.py

```diff
@@ -182,73 +194,76 @@ def stump(T_A, m, T_B=None, exclusion_zone=None, row_wise=False):
+            # self-join: right matrix profile index (top-1)
             if ignore_trivial and i < D.shape[0]:
-                IR = i + np.argmin(D[i:])  # shift argmin by `i` to get true index
+                IR = i + np.argmin(D[i:])  # shift arg by `i` to get true index
                 if D[IR] == np.inf:
                     IR = -1
-                I[i, 2] = IR
+                I[i, k + 1] = IR
 
     else:  # diagonal traversal
         if ignore_trivial:
             diags = np.arange(exclusion_zone + 1, n_A - m + 1)
         else:
             diags = np.arange(-(n_A - m + 1) + 1, n_B - m + 1)
 
-        for k in diags:
-            if k >= 0:
-                iter_range = range(0, min(n_A - m + 1, n_B - m + 1 - k))
+        for g in diags:
+            if g >= 0:
+                iter_range = range(0, min(n_A - m + 1, n_B - m + 1 - g))
             else:
-                iter_range = range(-k, min(n_A - m + 1, n_B - m + 1 - k))
+                iter_range = range(-g, min(n_A - m + 1, n_B - m + 1 - g))
 
             for i in iter_range:
-                D = distance_matrix[i, i + k]
-                if D < P[i, 0]:
-                    P[i, 0] = D
-                    I[i, 0] = i + k
+                d = distance_matrix[i, i + g]
+                if d < P[i, k - 1]:
+                    idx = searchsorted_right(P[i], d)
+                    # to keep the top-k, we must get rid of the last element.
+                    P[i, :k] = np.insert(P[i, :k], idx, d)[:-1]
+                    I[i, :k] = np.insert(I[i, :k], idx, i + g)[:-1]
 
                 if ignore_trivial:  # Self-joins only
-                    if D < P[i + k, 0]:
-                        P[i + k, 0] = D
-                        I[i + k, 0] = i
+                    if d < P[i + g, k - 1]:
+                        idx = searchsorted_right(P[i + g], d)
+                        P[i + g, :k] = np.insert(P[i + g, :k], idx, d)[:-1]
+                        I[i + g, :k] = np.insert(I[i + g, :k], idx, i)[:-1]
 
-                    if i < i + k:
+                    if i < i + g:
                         # Left matrix profile and left matrix profile index
-                        if D < P[i + k, 1]:
-                            P[i + k, 1] = D
-                            I[i + k, 1] = i
+                        if d < P[i + g, k]:
+                            P[i + g, k] = d
+                            I[i + g, k] = i
 
-                        if D < P[i, 2]:
+                        if d < P[i, k + 1]:
                             # right matrix profile and right matrix profile index
-                            P[i, 2] = D
-                            I[i, 2] = i + k
+                            P[i, k + 1] = d
+                            I[i, k + 1] = i + g
 
-    result = np.empty((l, 4), dtype=object)
-    result[:, 0] = P[:, 0]
-    result[:, 1:4] = I[:, :]
+    result = np.empty((l, 2 * k + 2), dtype=object)
+    result[:, :k] = P[:, :k]
+    result[:, k:] = I[:, :]
```

Comment:
I am indifferent. There is something aesthetically pleasing since the previous line is `P[:, :k]` but it's not necessary for `I` so I leave it up to you